From 5b9a6478f7046ad5c63ee0ef84e5efbbf37206dc Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Wed, 15 Mar 2023 16:28:22 +0200
Subject: [PATCH 3/5] s32r: fip_mem: Increase 'MAX_XLAT_TABLES' and
 'MAX_MMAP_REGIONS'

When booting with FIP in Mem support (XIP),
the 'MAX_XLAT_TABLES' should be increased, for S32R platforms.

As part of this commit, we increase 'MAX_XLAT_TABLES' for S32R
platforms in order to work in all scenarios.

Also, as a leftover from the original FIP in memory (XIP) support,
'MAX_MMAP_REGIONS' should also be increased by value one,
in case of S32R platforms, as it was done for S32G platforms.

Issue: ALB-9909
Upstream-Status: Pending

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32r/include/s32r_platform_def.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/nxp/s32/s32r/include/s32r_platform_def.h b/plat/nxp/s32/s32r/include/s32r_platform_def.h
index efb1c3599..3fa5c7739 100644
--- a/plat/nxp/s32/s32r/include/s32r_platform_def.h
+++ b/plat/nxp/s32/s32r/include/s32r_platform_def.h
@@ -21,8 +21,8 @@
 #define S32_MPIDR_CPU_MASK		0x1
 #define S32_MPIDR_CPU_MASK_BITS	0x1
 
-#define MAX_MMAP_REGIONS		32
-#define MAX_XLAT_TABLES			25
+#define MAX_MMAP_REGIONS		33
+#define MAX_XLAT_TABLES			26
 
 #if defined IMAGE_BL33
 #pragma warning "BL33 image is being built; you should configure it out."
-- 
2.25.1

